examples/14_ampere_tf32_tensorop_gemm/ampere_tf32_tensorop_gemm_cute.cpp

```diff
@@ -232,6 +236,24 @@ void test_gemm(int m, int n, int k)
   auto cute_result = std::vector<TC>(h_C.size());
   syclcompat::memcpy<TC>(cute_result.data(), d_C, h_C.size());
 
+#if defined(CUTLASS_ENABLE_MKL)
+  auto d_CRef = syclcompat::malloc<TC>(h_C.size());
+  auto d_hRef = std::vector<TC>(h_C.size());
+  auto queue = syclcompat::get_default_queue();
```

Comment:
what is the default queue in here, how can you make sure the queue is nvidia gpu queue. This test that you are running is specifically set for Amper device, and needed to be compiled only for that device due to definition of the architecture at compile time. how do you make sure that you get a correct queue in all systems when you use default here

Reply:
>  This test that you are running is specifically set for Amper device, and needed to be compiled only for that device due to definition of the architecture at compile time.

This example only builds if `SYCL_NVIDIA_TARGET` is defined ([here](https://github.com/codeplaysoftware/cutlass-fork/blob/091441f55fa62e666ec6a2d668358423bc547b40/examples/14_ampere_tf32_tensorop_gemm/CMakeLists.txt#L32)). 

> how can you make sure the queue is nvidia gpu queue

Since we do not explicitly create a queue, or pass any sort of device selection of to syclcompat, I would expect that the device selector is being set when calling the binary assuming multiple backends on the platform.

The reason I use the default_queue from sycl compat is because that's being used throughout, for memory allocations etc...

Reply:
The selector should be passed somehow probably through SYCLcompat, sycl default queue does not guaranty selection of any devices and you wont have any control when multiple GPU is available

Reply:
You can create a queue using the GPU device selector and then use `syclcompat::set_default_queue`. After that, you can add a check like the one in the cuda example.

```c++
  cudaDeviceProp props;

  cudaError_t error = cudaGetDeviceProperties(&props, 0);
  if (error != cudaSuccess) {
    std::cerr << "cudaGetDeviceProperties() returned an error: " << cudaGetErrorString(error) << std::endl;
    return -1;
  }

  if (!((props.major * 10 + props.minor) >= 80)) {
    std::cerr << "Ampere Tensor Core operations must be run on a machine with compute capability at least 80."
              << std::endl;
    notSupported = true;
  }

  if (notSupported) {
    // Returning zero so this test passes on older Toolkits. Its actions are no-op.
    return 0;
  }
```